spark/src/test/scala/org/apache/spark/sql/delta/schema/InvariantEnforcementSuite.scala

```diff
@@ -167,54 +167,163 @@ class InvariantEnforcementSuite extends QueryTest
     }
   }
 
-  test("write empty DataFrame - zero columns") {
```

Comment:
I removed this test since all it was really testing was that a not null exception was thrown before a can't write an empty dataframe exception. This is a weird edge case that would make the code even more complicated, and an exception is thrown either way so it doesn't seem very useful